Fatras/include/ActsFatras/EventData/Particle.hpp

```diff
@@ -163,10 +164,13 @@ class Particle {
   /// Particle mass.
   constexpr Scalar mass() const { return m_mass; }
 
+  /// Particle hypothesis.
+  constexpr Acts::ParticleHypothesis hypothesis() const {
```

Comment:
in this implementation the `Particle` is more general than the `ParticleHypothesis` as we strip off the sign of pdg and charge. We could add a `hypothesis` member to avoid the creation of temporaries but otherwise I do not see how to embed this further